rope/base/utils.py

```diff
@@ -81,3 +81,48 @@ def __call__(self, *args, **kwds):
         if len(self.cache) > self.count:
             del self.cache[0]
         return result
+
+class memoize(object):
```

Comment:
As I see, class memoize has same responsibility as class _Cached (but with minor higher performance). We don't need two classes for same purpose. We can improve _Cached class (use dict instead list, etc.) and remove class memoize.